### PR TITLE
bumping kapp-controller's package to 0.34.0

### PR DESCRIPTION
**Pull request by ridavid2002**
### What this PR does / why we need it

- Bump kapp-controller to 0.34.0

### Which issue(s) this PR fixes

Fixes #
TKG-9983

### Release note

```
Bump kapp-controller to 0.34.0
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.33.1/bundle/config/values.star [hunk 1]
```diff
@@ -14,5 +14,5 @@ def generateBashCmdForDNS(coreDNSIP):
     # In this way, Kapp Controller will have cluster IP access,
     # and still able to resolve enternal urls when core DNS is unavailable
 
-    return "cp /etc/resolv.conf /etc/resolv.conf.bak; sed '1 i nameserver " + coreDNSIP + "' /etc/resolv.conf.bak > /etc/resolv.conf; rm /etc/resolv.conf.bak; cp -R /etc/* /kapp-etc; chmod g+w /kapp-etc/pki/tls/certs/ca-bundle.crt && chgrp 2000 /kapp-etc/pki/tls/certs/ca-bundle.crt"
+    return "cp /etc/resolv.conf /etc/resolv.conf.bak; sed '1 i nameserver " + coreDNSIP + "' /etc/resolv.conf.bak > /etc/resolv.conf; rm /etc/resolv.conf.bak; cp -R /etc/* /kapp-etc; chmod g+w /kapp-etc/pki/tls/certs/ca-bundle.crt"
```
Comment (shyaamsn) [inline pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.33.1/bundle/config/values.star:17]: @ridavid2002 I think this bug fix https://github.com/vmware-tanzu/community-edition/pull/3259/files  
is missed while adding 0.33.1 kapp-controller package in TCE. Could you open a PR in TCE to fix this. Thanks

Comment (shyaamsn) [reply pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.33.1/bundle/config/values.star:17]: @aaronshurley Is the chgrp 2000 /kapp-etc/pki/tls/certs/ca-bundle.crt intentionally removed ?

Comment (neil-hickey) [reply pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.33.1/bundle/config/values.star:17]: yep! This conversation might make things clearer on that decision: https://github.com/vmware-tanzu/community-edition/pull/3476#discussion_r823163912

**Comment by shivaani0505** [comment]
Please remove TKG-9983 from Fixes# and refer to issue created for TF for this bump

File: pkg/v1/providers/vendir.yml.save [hunk 2]
```diff
@@ -0,0 +1,27 @@
+apiVersion: vendir.k14s.io/v1alpha1
```
Comment (shivaani0505) [inline pkg/v1/providers/vendir.yml.save:1]: remove .save file

Comment (ridavid2002) [reply pkg/v1/providers/vendir.yml.save]: this has been removed

**Comment by ridavid2002** [comment]
//usebom:sandbox/v1.6.0-zshippable/400944503990521449/tkg-compatibility

**Review by vijaykatam — APPROVED** [review]

**Comment by jayunit100** [comment]
hmmm 
- tests failing
- marked as ok to merge ...
are we going to YOLO merge this ? or wait for CAPZ issues to get fixed?

**Comment by shyaamsn** [comment]
> hmmm
> 
> * tests failing
> * marked as ok to merge ...
>   are we going to YOLO merge this ? or wait for CAPZ issues to get fixed?
> 
> ... also, howcome no `usebom` directive in this mr @shyaamsn (i think it got deleted)?

I deleted the usebom comment to see if it makes any difference w.r.t TKG CAPZ integration test. It didnt make any difference which means the issue is already present.

**Comment by ridavid2002** [comment]
//usebom:sandbox/v1.6.0-zshippable/401219495596661665/tkg-compatibility

**Comment by ridavid2002** [comment]
//usebom:sandbox/v1.6.0-zshippable/401220038440257848/tkg-compatibility

**Comment by ridavid2002** [comment]
//usebom:sandbox/v1.6.0-zshippable/401219495596661665/tkg-compatibility

**Review by jayunit100 — APPROVED** [review]

**Pull request merged**